mlir/lib/Dialect/Bufferization/Transforms/EmptyTensorElimination.cpp

```diff
@@ -152,6 +152,9 @@ LogicalResult mlir::bufferization::eliminateEmptyTensors(
       if (emptyTensorOp == replacement.getDefiningOp())
         continue;
       if (replacement.getType() != v.getType()) {
+        if (ShapeAdaptor(replacement.getType()).getElementType() !=
```

Comment:
We usually write `cast<ShapedType>(replacement.getType()).getElementType()` instead of `ShapeAdaptor`.

Reply:
Thanks for the advice! I have changed here accordingly.